sdk/core/core/inc/az_span.h

```diff
@@ -25,7 +25,7 @@ typedef struct {
 
 typedef int32_t az_result_byte;
 
-extern az_span const AZ_SPAN_NULL;
+AZ_NODISCARD AZ_INLINE az_span az_span_create_empty() { return (az_span){ 0 }; }
```

Comment:
probably not needed, just have people write the zero span.

Reply:
I really like AZ_SPAN_NULL; What if this were a define like NULL (since you don't like global const)?

Reply:
@JeffreyRichter as far as I understood, there are two questions:
1. null vs empty. We are already have an empty span and currently it's a span that has `size == 0`. For example, we have a function `az_span_is_empty()`. Usually, NULL applies for pointers. A span is not a pointer (like C# string) but it's a pair of begin/end values. For example, a C++ range or a Rust slice don't have NULL range. We can introduce a special value but I'm afraid it can be similar to a Tony Hoare's billion-dollar mistake.
2. Use #define instead of inline function. As far as I remember, we should use inline functions and enum values where we can instead of macros. @barcharcraz may comment on it.

Reply:
@barcharcraz Usually, we use `{ 0 }` to initialize it to a deterministic value, for example, before calling `_init` method. We also are trying to design our types that `{ 0 }` means a default valid value. But I'm ok with having empty/null constants/inline functions if people would like to use it to clearly specify a meaning of the value, even if the value is `{ 0 }`.

Reply:
BTW, new C# Span is a value type and can't be NULL https://docs.microsoft.com/en-us/dotnet/api/system.span-1?view=netcore-3.0

Reply:
In case we wanted a constant, I'd prefer `(az_span){ 0 }` to `AZ_SPAN_NULL`